File: app/src/main/java/com/libertystudio/bookshop/fragment/BooksFragment.java
package com.libertystudio.bookshop.fragment;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListView;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.libertystudio.bookshop.MainActivity;
import com.libertystudio.bookshop.R;
import com.libertystudio.bookshop.data.BookAdapter;

public class BooksFragment extends BaseFragment {
    private ListView lvBooks;

    private MainActivity mainActivity;

    public BooksFragment() { }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_books, container, false);
        initElements(view);

        return view;
    }

    private void initElements(View view) {
        mainActivity = (MainActivity) getActivity();

        lvBooks = view.findViewById(R.id.lv_books);
        lvBooks.setAdapter(new BookAdapter(mainActivity, mainActivity.getListBooks()));
        lvBooks.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                mainActivity.setSelectedBook(mainActivity.getListBooks().get(position));
                mainActivity.setTitle("О книге");

                Fragment fragmentBookInfo = new BookInfoFragment();
                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
                fragmentTransaction.replace(R.id.content, fragmentBookInfo).commit();
            }
        });
    }
}

File: app/src/main/java/com/libertystudio/bookshop/fragment/BooksFragment.kt
package com.libertystudio.bookshop.fragment

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import com.libertystudio.bookshop.MainActivity
import com.libertystudio.bookshop.R
import com.libertystudio.bookshop.adapter.BaseAdapterCallback
import com.libertystudio.bookshop.adapter.BookAdapter
import com.libertystudio.bookshop.entity.Book
import kotlinx.android.synthetic.main.fragment_books.*

class BooksFragment : BaseFragment() {
    private var mainActivity: MainActivity? = null

    private val bookAdapter = BookAdapter()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_books, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        mainActivity = activity as MainActivity?

        initAdapter()
        initView()
    }

    private fun initAdapter() {
        bookAdapter.setDataList(mainActivity!!.listBooks)
        bookAdapter.callback = object : BaseAdapterCallback<Book> {
            override fun onItemClick(item: Book) {
                startFragment(BookInfoFragment.newInstance(item))
            }
        }
    }

    private fun initView() {
        setTitle(resources.getString(R.string.fragment_books_title))

        rvBooks?.apply {
            layoutManager = LinearLayoutManager(context)
            adapter = bookAdapter
        }
    }
}